src/TestFramework/Coverage/XmlReport/TestTrace.php

```diff
@@ -37,29 +37,38 @@
 
 use function array_key_exists;
 use Infection\AbstractTestFramework\Coverage\CoverageLineData;
-use Infection\TestFramework\Coverage\CoverageReport;
-use Infection\TestFramework\Coverage\LineCodeCoverage;
 use Infection\TestFramework\Coverage\NodeLineRangeData;
+use Infection\TestFramework\Coverage\TestLocations;
+use Infection\TestFramework\Coverage\Trace;
 
 /**
+ * Impartial trace providing information regarding the tests but completely lacks any awareness of
+ * its associated source files.
+ *
+ * TODO: FileMutationGenerator::generate() ends up having to rely on both SourceFileData & Trace due
+ *  to the fact that a Trace right now only exposes a test-related API
+ *  (`hasTests()`, `getAllTestsForMutation()`). Maybe a cleaner solution would be expose the
+ *  associated source file API to `Trace` as well, i.e. moving up `SourceFileData::getSplFileInfo()`
+ *  to `Trace`. This current class `TestsTrace` could then become a simple helper for `SourceFileData`
+ *  instead of a full-pledge `Trace` implementation. And FileMutationGenerator as a result would
+ *  also be simplified to not have to rely on SourceFileData which is `Trace` implementation and
+ *  depend on `Trace` only.
```

Comment:
Big TODO

Reply:
I'd argue merging them together will cause a conflict of concerns. The original Trace object has nothing to do with mutated files. As an alternative approach this proxy could be made a full-fledged trace provider, with hopefully minimal changes to the consumers.

Reply:
IIRC the reason this is proxy is convenience sake, no more.